New submission - Cacao theme from a new contributor

Michael-Garibaldi opened the pull request:
Adding in a new theme for consideration

This is a theme called Cacao.  It uses a slightly larger font and uses a gentler color scheme for OLDER eyes, like mine
This is my first package ever submitted, so there will very likely be errors or issues. I'm happy to fix any that you find.
I have a screenshot on my github to show what this theme looks like and how it came to be.



### Direct link to the package repository

https://github.com/Michael-Garibaldi/Cacao-theme

### Your association with the package

I'm the maintainer and the nearly sole contributor. A bug fix was found and the gentleman was credited in the source code

### Relevant communications with the upstream package maintainer

**None Needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this. I believe this file is misformatted (you can view other files for reference).  Also your recipe should be/contain a lower-cased package name.  Hence the following -

⸺ `cacao-theme.el` with byte-compile using Emacs 30.1:
```
cacao-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
```

⸺ `cacao-theme.el` with package-lint 20250819.1304:
```
7 issues found:
1:0: error: Package should have a ;;; Commentary section.
1:0: error: package.el cannot parse this buffer: Package lacks a file header
9:8: error: Package URLs should be a single HTTPS or HTTP URL.
10:20: warning: "" is not a valid version. MELPA will handle this, but other archives will not.
165:0: warning: Closing parens should not be wrapped onto new lines.
183:2: error: You should depend on (emacs "24.1") if you need `org-date'.
202:2: error: You should depend on (emacs "24.1") if you need `org-priority'.
```

⸺ `cacao-theme.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- Warning: No `add-to-list 'custom-theme-load-path` found
```

⸺ `cacao-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
cacao-theme.el:0: You should have a section marked ";;; Commentary:"
cacao-theme.el:39: You should have a section marked ";;; Code:"
cacao-theme.el:218: The footer should be: (provide 'cacao-theme)\n;;; cacao-theme.el ends here
cacao-theme.el:39: First sentence should end with punctuation
cacao-theme.el:5: White space found at end of line
```

Michael-Garibaldi commented:
@riscy 

I just made the changes you requested. The error :

9:8: error: Package URLs should be a single HTTPS or HTTP URL.

surprises me as that's a comment and as far as I knew, Elisp doesn't process those.  I changed it to match the other URL I saw in the code so it should work now.

Thanks for the feedback!

riscy commented:
> that's a comment and as far as I knew

Your current boilerplate:
```elisp
;; Author: Panaman Creel 
;; modeline correction and color class by RickMMA@github.com
;; URL: 
;; Package-Version: 
;; Package-Commit: 
;; Version: 
```
The package `URL:` above should not be blank - it should point somewhere.  You should fill these details in and run package-lint on your code.

The other existing `package-lint` warnings and errors are legitimate for this package too - if you need help addressing a specific one, let me know.  It's okay to take some time on this - there's no rush.

Michael-Garibaldi commented:
> Thanks for the update! Just checking in: it looks like there are no elisp files in your repository's root at the moment.

I just corrected that.   Sorry!

Michael-Garibaldi commented:
> Thank you! Here's the latest -
> 
> ⸺ `cacao-test-theme.el` with byte-compile using Emacs 30.1:
> 
> ```
> cacao-test-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
> cacao-test-theme.el:63:4: Warning: docstring wider than 80 characters
> ```
> 
> ⸺ `cacao-test-theme.el` with [melpazoid](https://github.com/riscy/melpazoid):
> 
> ```
> - cacao-test-theme.el#L61: No space between the function name and argument list
> ```
> 
> ⸺ `cacao-test-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> cacao-test-theme.el:21: You should have a section marked ";;; Code:"
> cacao-test-theme.el:22: First sentence should end with punctuation
> cacao-test-theme.el:3: White space found at end of line
> ```
> 
> `#'load`-check on each file:
> 
> ```
> Loading cacao-test-theme.el
>   cacao-test-theme.el:Error: Emacs 30.1:
>   (error "Unknown theme `cacao'")
> ```
> 
> ⸺ Package and license:
> 
> * No 'main' file found, e.g. 'cacao-theme.el'
> * Can't check Package-Requires if there is no 'main' file
> * Add a LICENSE file to the repository
>   See: https://github.com/licensee/licensee
> * cacao-test-theme.el needs _formal_ license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)
> * MELPA requires a file called cacao-theme.el
> * cacao-test-theme.el -- not in package namespace `cacao-theme-`

WOOPS!  That was the test version where all my changes are tested first.  I just uploaded cacao-theme.el.  I checked it with the linter and it shows "No issues found".  Sorry about that!

riscy commented:
I think there are no package-lint errors right now, which is why you're not seeing them - those are byte-compile and checkdoc (separate workflows)

Michael-Garibaldi commented:
@riscy

OK, I just ran the tab through all lines, ran the linter and ran the
byte-compile program and checkdoc (both in 30.1 and 29) and both compile
just fine, no errors, and checkdoc shows
Buffer comments and tags: Ok
Documentation style: Ok
Message / Query text style: Ok
Unwanted Spaces: OK

Looks ok as of right now.

On Sun, Sep 21, 2025 at 1:58 PM Chris Rayner ***@***.***>
wrote:

> *riscy* left a comment (melpa/melpa#9576)
> <https://github.com/melpa/melpa/pull/9576#issuecomment-3316156832>
>
> I think there are no package-lint errors right now, which is why you're
> not seeing them - those are byte-compile and checkdoc (separate workflows)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/melpa/melpa/pull/9576#issuecomment-3316156832>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BO3Y2UTV7U3ODLGYJFVCOL33T3RNZAVCNFSM6AAAAACEDQUMNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMJWGE2TMOBTGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

riscy commented:
Thanks again for accommodating.  As a reminder -

- Add a LICENSE file to the repository
  See: https://github.com/licensee/licensee
- cacao-theme.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

Michael-Garibaldi commented:
> Thanks again for accommodating. As a reminder -
> 
> * Add a LICENSE file to the repository
>   See: https://github.com/licensee/licensee
> * cacao-theme.el needs _formal_ license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

@riscy ,

No problem!  You're the boss for this pull request  :)

Anyrate, I added in the License file into my github and added in a license boilerplate into my theme.
I also ran it through a linter, checkdoc and byte-compile. All went through just fine.

Thanks again for you time!

riscy commented:
Thank you!  I will merge this now with just a few minor issues for you to sort out when you have time -

Re: the license tag,
```
;; License: GPL-3+
```
Consider an SPDX license identifier for this line instead, like `;; SPDX-License-Identifier: GPL-3.0-or-later`, which can be formally recognized by tooling.

Your summary line `Based on a image reversal` should be `Based on an image reversal` but I think `Theme based on a color-inverted image` would be clearer, at least to me.

In your `;;; Commentary` - I'd much rather you expanded on your "image reversal" inspiration, whether this is a light or dark theme, what colors are prominent, etc, than give me a link to https://imagecolorpicker - people browsing the index for a color theme will be more interested in what the color theme is than how it was made.

All that said, I'll merge.  Welcome to MELPA! :)
